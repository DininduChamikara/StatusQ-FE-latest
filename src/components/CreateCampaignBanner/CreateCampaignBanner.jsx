/* eslint-disable jsx-a11y/alt-text */
import { AddCircleOutline, Campaign } from "@mui/icons-material";
import { Box, Button, Paper, Stack, Typography } from "@mui/material";
import React from "react";
import CampaignImage from "../../images/campaigns.jpg";

function CreateCampaignBanner({ setVisibleStepper }) {
  return (
    <Box data-testid="createCampaignBanner-1">
      <Paper
        elevation={3}
        sx={{
          p: 3,
          width: "100%",
          display: { xs: "block", lg: "flex" },
          alignItems: "center",
          flexDirection: "row",
          justifyContent: "center",
        }}
      >
        <Box sx={{ width: { xs: "100%", lg: "50%" } }}>
          <img height={300} src={CampaignImage} />
        </Box>
        <Box
          sx={{ width: { xs: "100%", lg: "50%" }, alignItems: "center", p: 2 }}
        >
          <Typography
            sx={{
              fontSize: "1.8rem",
              fontWeight: "bold",
              my: 2,
              color: "primary.darker",
            }}
          >
            Promote your Products or Services on Social Media
          </Typography>
          <Typography sx={{ color: "primary.dark" }}>
            Click on Create New Campaigns to launch an advertisement campaign on
            a Social Media Platform for your product or service
          </Typography>
          <Box sx={{ my: 2 }}>
            <Stack direction="row" spacing={2}>
              <Button
                onClick={() => {
                  setVisibleStepper(true);
                }}
                variant="contained"
                color="primary"
                startIcon={<AddCircleOutline />}
              >
                Create New Campaign
              </Button>
              <Button
                onClick={() => {
                  setVisibleStepper(false);
                }}
                variant="contained"
                color="secondary"
                endIcon={<Campaign />}
              >
                Ongoing Campaigns
              </Button>
            </Stack>
          </Box>
        </Box>
      </Paper>
    </Box>
  );
}

export default CreateCampaignBanner;
