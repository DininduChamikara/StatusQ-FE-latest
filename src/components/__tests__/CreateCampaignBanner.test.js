import { render, screen, cleanup } from "@testing-library/react";
import renderer from 'react-test-renderer';
import CreateCampaignBanner from "../CreateCampaignBanner/CreateCampaignBanner";

afterEach(() => {
    cleanup();
})

test("should render CreateCampaignBanner", () => {
  render(<CreateCampaignBanner />);
  const createCampaignBannerElement = screen.getByTestId(
    "createCampaignBanner-1"
  );
  expect(createCampaignBannerElement).toBeInTheDocument();
  expect(createCampaignBannerElement).toHaveTextContent(
    "Promote your Products or Services on Social MediaClick on Create New Campaigns to launch an advertisement campaign on a Social Media Platform for your product or serviceCreate New CampaignOngoing Campaigns"
  );
});

test('matches snapshot', () => {
    const tree = renderer.create(<CreateCampaignBanner/>).toJSON();
    expect(tree).toMatchSnapshot();
})
