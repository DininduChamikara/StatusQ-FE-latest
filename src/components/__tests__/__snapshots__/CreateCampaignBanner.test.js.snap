// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiBox-root css-0"
  data-testid="createCampaignBanner-1"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-2plwtp-MuiPaper-root"
  >
    <div
      className="MuiBox-root css-1uqxghp"
    >
      <img
        height={300}
        src="campaigns.jpg"
      />
    </div>
    <div
      className="MuiBox-root css-14a5gor"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-bby0vw-MuiTypography-root"
      >
        Promote your Products or Services on Social Media
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-gtcsjv-MuiTypography-root"
      >
        Click on Create New Campaigns to launch an advertisement campaign on a Social Media Platform for your product or service
      </p>
      <div
        className="MuiBox-root css-1ebprri"
      >
        <div
          className="css-e53awj-MuiStack-root"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddCircleOutlineIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
            </span>
            Create New Campaign
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Ongoing Campaigns
            <span
              className="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CampaignIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 11v2h4v-2h-4zm-2 6.61c.96.71 2.21 1.65 3.2 2.39.4-.53.8-1.07 1.2-1.6-.99-.74-2.24-1.68-3.2-2.4-.4.54-.8 1.08-1.2 1.61zM20.4 5.6c-.4-.53-.8-1.07-1.2-1.6-.99.74-2.24 1.68-3.2 2.4.4.53.8 1.07 1.2 1.6.96-.72 2.21-1.65 3.2-2.4zM4 9c-1.1 0-2 .9-2 2v2c0 1.1.9 2 2 2h1v4h2v-4h1l5 3V6L8 9H4zm11.5 3c0-1.33-.58-2.53-1.5-3.35v6.69c.92-.81 1.5-2.01 1.5-3.34z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
